vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,343 @@
+    }
+    else if (CIRCLE_COLOR.equals(key)) {
+      return new StringBuilder().append(ColorConfigurer.colorToString(this.color)).toString();
+    }
+    else if (PULSES.equals(key)) {
+      return new StringBuilder().append(this.pulses).toString();
+    }
+    else if (PULSES_PER_SEC.equals(key)) {
+      return new StringBuilder().append(this.pulsesPerSec).toString();
+    }
+    return null;
+  }
+  
+  
+  public void setAttribute(String key, Object value) {
+    if (FLARE_KEY.equals(key)) {
+      if (FLARE_ALT_LOCAL.equals(value)) {
+        this.flareKey = FLARE_ALT;
+      } 
+      else if (FLARE_COMMAND_LOCAL.equals(value) || FLARE_CTRL_LOCAL.equals(value)) {
+        this.flareKey = FLARE_CTRL;
+      } 
+      else {
+        this.flareKey = (String) value;
+      }
+    }
+    else if (CIRCLE_SIZE.equals(key)) {
+      if (value instanceof String) {
+        this.circleSize = Integer.parseInt((String) value); 
+      } 
+      else if (value instanceof Integer) {
+        this.circleSize = (Integer) value;
+      }
+    } 
+    else if (CIRCLE_COLOR.equals(key)) {
+      if (value instanceof String) {
+        value = ColorConfigurer.stringToColor((String) value);
+      }
+      this.color = (Color)value;
+    }
+    else if (PULSES.equals(key)) {
+      if (value instanceof String) {
+        this.pulses = Integer.parseInt((String) value);
+      } 
+      else {
+        this.pulses = (Integer) value;
+      }
+    }
+    else if (PULSES_PER_SEC.equals(key)) {
+      if (value instanceof String) {
+        this.pulsesPerSec = Integer.parseInt((String) value);
+      } 
+      else {
+        this.pulsesPerSec = (Integer) value;
+      }
+    }
+  }
+  
+  
+  @Override
+  public HelpFile getHelpFile() {
+    return HelpFile.getReferenceManualPage("Flare.htm");  //$NON-NLS-1$
+  }
+  
+
+  public void addTo(final Buildable parent) {
+    if (parent instanceof Map) {
+      this.map = (Map) parent;
+      GameModule.getGameModule().addCommandEncoder((CommandEncoder) this);
+      this.map.addDrawComponent((Drawable) this);
+      this.map.addLocalMouseListener((MouseListener) this);
+      this.timer  = new Timer(CIRCLE_RATE, this);
+      this.frame  = 0;
+    }
+  }
+  
+  public void removeFrom(final Buildable parent) {
+    if (parent instanceof Map) {
+      GameModule.getGameModule().removeCommandEncoder((CommandEncoder) this);
+    }
+  }
+
+
+  public void startAnimation(final boolean isLocal) {
+    if (!isLocal) {
+      this.map.centerAt(this.clickPoint);
+    }
+    this.active = true;
+    this.timer.restart();
+    this.frame = 0;
+    
+    this.animate = ((this.pulses > 0) && (this.pulsesPerSec > 0));
+    if (this.animate) {
+      this.framesPerPulse = 30 / this.pulsesPerSec;
+      if (this.framesPerPulse < 1) this.framesPerPulse = 1;
+      this.frames = this.pulses * framesPerPulse;
+    } 
+    else {
+      this.frames = DEFAULT_FRAMES;
+    }
+    
+    this.map.getView().repaint();
```

Comment:
You could limit the repaint just to the dirty area.

Reply:
I am interested in doing that - can you point me to some sample/related code?

Reply:
There's more than one way you can do it. If you have view coordinates handy, you can call the four-argument version of `repaint(x, y, w, h)`,

If you have map coordinates handy, which you do, you can do this:
```
map.repaint(new Rectangle(
  clickPoint.x - circleSize / 2 - 3,
  clickPoint.y - circleSize / 2 - 3,
  circleSize + 6,
  circleSize + 6
));
```
That includes a bit of padding for the stroke width. (It _shouldn't_ extend 3px outward in all directions, given that it's only got a width of 3.)